configure.ac

```diff
@@ -88,21 +88,19 @@ AM_CONDITIONAL(MPI_ENGINE, test x$mpi_engine = xyes)
 
 if test x$mpi_engine = xyes; then
   AC_DEFINE(MPI_ENGINE, 1, MPI Engine enabled)
-  AC_CHECK_PROG(mpicxx_present, mpicxx, [yes], [no])
-  if test x$mpicxx_present = xno; then
-    AC_CHECK_PROG(mpicxx_present, mpiCC, [yes], [no])
-    if test x$mpicxx_present = xno; then
-      AC_MSG_ERROR([Fatal Error: no mpi compiler found])
-      exit;
-    else
-      CXX=mpiCC
-    fi
-  else
-    CXX=mpicxx
-  fi
+  AC_CHECK_PROGS(MPICXX, mpicxx mpic++ CC, $MPICXX)
```

Comment:
This is the main change of this PR.
The rest is just a bunch of files changed because of this change.

Reply:
I see